### ISSUE TITLE
Memory monitoring shows incorrect memory usage. 

### ISSUE THREAD

**Issue by gabriel-andersson** [ray-project/ray#33741]
### What happened + What you expected to happen

When running distributed training using `ray.train.torch.TorchTrainer`, deployed with kuberay in a kubernetes cluster, the nodes are eventually killed by the memory monitoring because "memory usage" > `RAY_memory_usage_threshold` (95%). The dashboard is also displaying this memory usage:
 
![dashboard](https://user-images.githubusercontent.com/111285127/227972258-bdb1f9cd-6c44-4daf-a694-92ce81517909.png)

However, the actual memory usage on the nodes when looking at `top`, `free` and `ray memory` in the node shell are closer to 20%. Plotting the result from running `psutil.virutal_memory()` from a python session in the node shell yields:
```
MEMORY
------
Total      :   54.9G
Available  :   45.4G
Percent    :    17.4
Used       :    8.0G
Free       :  926.4M
Active     :   42.9G
Inactive   :    7.3G
Buffers    :  743.9M
Cached     :   45.3G
Shared     :  961.2M
Slab       :    3.4G

SWAP
----
Total      :    0.0B
Used       :    0.0B
Free       :    0.0B
Percent    :     0.0
Sin        :    0.0B
Sout       :    0.0B
```
It seems like the memory monitoring interprets the "Cached memory" as "Used" instead of "Available". Looking at the function,
```
def _get_mem_usage():
        total = ray._private.utils.get_system_memory()
        used = ray._private.utils.get_used_memory()
        available = total - used
        percent = round(used / total, 3) * 100
        return total, available, percent, used
```
it seems like the RAM cache is indeed interpreted as "used" when entering ` ray._private.utils.get_used_memory()` and going into the `get_cgroupv1_used_memory(filename) `.

### Versions / Dependencies

Docker base image: 
rayproject/ray:2.3.0-py310-cpu

torch==1.13.1
torchvision==0.14.1

### Reproduction script

-

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

**Comment by rkooo567** [comment]
After the initial discussion, it looks like we are including the cached memory when calculating the cgroup1 mem usage. Since users have the high cached memory, it causes the issue Cached     :   45.3G

Given cached memory is freed whenever additional mem is required, we should probably not include when calculating memory

**Comment by jjyao** [comment]
Fixed by #43071
